src/zkp/whitelist.rs

```diff
@@ -0,0 +1,470 @@
+//! Bindings for the "whitelist" ring signature implementation in secp256k1-zkp.
+//!
+//! This implementation is used for Liquid PAK list inclusion proofs.
+
+#[cfg(feature = "std")]
+use std::{fmt, str};
+
+use core::ptr;
+
+use ffi::CPtr;
+#[cfg(feature = "std")]
+use from_hex;
+use {ffi, secp256k1, Error, PublicKey, Secp256k1, SecretKey, Signing, Verification};
+
+/// A whitelist ring signature.
+#[derive(Clone, PartialEq, Eq, Hash)]
+#[repr(transparent)]
```

Comment:
Does this `repr` allow a cast via `as` from `ffi::WhitelistSignature`? In that case, I think we should not have `Default` on `ffi::WhitelistSignature` but instead have an unsafe constructor.

Alternatively, we can remove `repr` here. What do we need it for?